packages/salesforce-adapter/test/change_validators/flow_referenced_elements.test.ts

```diff
@@ -0,0 +1,163 @@
+    },
+    fields: {
+      targetReference: { refType: BuiltinTypes.STRING },
+    },
+  })
+  const FlowNode = createMetadataObjectType({
+    annotations: {
+      metadataType: 'FlowNode',
+    },
+    fields: {
+      name: { refType: BuiltinTypes.STRING },
+      locationX: { refType: BuiltinTypes.NUMBER, annotations: { constant: 1 } },
+      locationY: { refType: BuiltinTypes.NUMBER, annotations: { constant: 1 } },
+      connector: { refType: FlowConnector, annotations: { required: false } },
+    },
+  })
+  const Flow = createMetadataObjectType({
+    annotations: {
+      metadataType: 'Flow',
+    },
+    fields: {
+      start: { refType: FlowNode },
+      actionCalls: { refType: new ListType(FlowNode), annotations: { required: false } },
+      assignments: { refType: new ListType(FlowNode), annotations: { required: false } },
+      decisions: { refType: new ListType(FlowNode), annotations: { required: false } },
+      recordCreates: { refType: new ListType(FlowNode), annotations: { required: false } },
+    },
+  })
+  describe('when all flow elements are existing and referenced', () => {
+    beforeEach(() => {
+      const flow = createInstanceElement(
+        {
+          fullName: 'TestFlow',
+          start: {
+            connector: { targetReference: 'ActionCall' },
+          },
+          actionCalls: [
+            {
+              name: 'ActionCall',
+              connector: { targetReference: 'Assignment' },
+            },
+          ],
+          assignments: [
+            {
+              name: 'Assignment',
+              connector: { targetReference: 'Decision' },
+            },
+          ],
+          decisions: [
+            {
+              name: 'Decision',
+              connector: { targetReference: 'RecordCreate' },
+            },
+          ],
+          recordCreates: [
+            {
+              name: 'RecordCreate',
+            },
+          ],
+        },
+        Flow,
+      )
+      flowChange = toChange({ after: flow })
+    })
+    it('should not return any errors', async () => {
+      const errors = await flowReferencedElements([flowChange])
+      expect(errors).toHaveLength(0)
+    })
+  })
+  describe('when there are references to missing flow elements', () => {
+    beforeEach(() => {
+      const flow = createInstanceElement(
+        {
+          fullName: 'TestFlow',
+          start: {
+            connector: { targetReference: 'ActionCall' },
+          },
+        },
+        Flow,
+      )
+      flowChange = toChange({ after: flow })
+    })
+    it('should not return any errors', async () => {
+      const errors = await flowReferencedElements([flowChange])
+      expect(errors).toHaveLength(1)
+      const [error] = errors
+      expect(error.severity).toEqual('Error')
+      expect(error.message).toEqual('Reference to missing Flow Element')
+      expect(error.detailedMessage).toEqual(`The Flow Element "${'ActionCall'}" does not exist.`)
```

Comment:
```suggestion
      expect(errors).toEqual([{
        message: '...',
        detailedMessge: '...',
        severity: 'Error',
        elemID: missingRefElemID,
      }])
```
Nit: Can be greatly simplified